src/fetchers.ts

```diff
@@ -131,7 +154,70 @@ async function fetchUrl(url: string, context: FetchContext): Promise<string> {
   }
 
   if (body === undefined) {
-    const response = await fetch(url);
+    if (spec.private && !context.tokens) {
+      // if no tokens and the repo is private, fallback to git ls-remote
+      const p = Deno.run({
+        cmd: [
+          "git",
+          "ls-remote",
+          `git@github.com:${spec.owner}/${spec.repo}`,
+          spec.ref ?? "HEAD",
+        ],
+        stdout: "piped",
+      });
```

Comment:
I added this fallback so that you can still do all the checking without needing a github token for private repositories. Not sure if it would have been better to just try via API first and then fallback instead of requiring `private` to be set.